src/libostree/ostree-sysroot.c

```diff
@@ -1324,10 +1325,36 @@ ostree_sysroot_repo (OstreeSysroot *self)
   return self->repo;
 }
 
+static OstreeBootloader*
+_ostree_sysroot_new_bootloader_by_type (
+    OstreeSysroot                 *sysroot,
+    OstreeCfgSysrootBootloaderOpt  bl_type)
+{
```

Comment:
It is my understanding that this function (judging by the calling logic) is not meant to be reached with an `AUTO` bootloader setting, despite using the `OstreeCfgSysrootBootloaderOpt` type which allows that.
If so, it would be good having an explicit assertion (outside of the `default` catch-all switch case) to call out that special case. It would also help casual readers landing here.

Reply:
I've added a `case` and a comment.